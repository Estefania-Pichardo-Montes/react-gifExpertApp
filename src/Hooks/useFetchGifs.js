import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifts";

//Este es un custom hook
//Funcionan como si fueran functional components

export const useFetchGift=(Category)=>{
    const [state, setstate] = useState({
            data:[],
            loading:true
     });

    useEffect(()=>{
        getGifs(Category).then(imgs=>{

            setstate({
              data:imgs,
              loading:false
          })       
        });
    },[Category])

    return state;
}