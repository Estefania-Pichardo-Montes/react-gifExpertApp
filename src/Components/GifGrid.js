import React from 'react'
import { useFetchGift } from '../Hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({Category}) => {

    const { data:images, loading } = useFetchGift(Category);

    return (
        <>
            <h3 className="animate__animated fadeIn">{Category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading</p>}
             <div className="cardGrid">
           
                {
                    images.map((img)=>
                        <GifGridItem 
                            key={img.id}
                            {...img}
                        />
                    )
                }
             </div>
        </>

    )
}
GifGrid.propTypes={
    Category:PropTypes.string.isRequired
}