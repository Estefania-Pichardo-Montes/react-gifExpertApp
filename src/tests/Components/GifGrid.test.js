import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../Components/GifGrid';
import { useFetchGift } from '../../Hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../../Hooks/useFetchGifs') //Finje la llamada a este archivo para controlar la informacion que va a responder

describe('Pruebas en el gifGrid', () => {

    const category='Green Arrow';

    test('Debe hacer match con el snapshot', () => {
       useFetchGift.mockReturnValue({
           data:[],
           loading:true
       });

        const wrapper=shallow(<GifGrid Category={category}/>); 
       expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar items cuando se cargan imagenes  useFetchGifts', () => {
       const gifs=[{
           id:'ABC',
           url:'https://localhost/cualquier/cosa.jpg',
           title:'Cualquier cosa'
       },
       {
        id:'ABC',
        url:'https://localhost/cualquier/cosa.jpg',
        title:'Cualquier cosa'
       }
    
        ];

       useFetchGift.mockReturnValue({
            data:gifs,
            loading:false
        });

        const wrapper=shallow(<GifGrid Category={category}/>); 

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
})
