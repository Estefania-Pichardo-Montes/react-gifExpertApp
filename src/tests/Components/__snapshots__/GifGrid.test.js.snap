// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el gifGrid Debe hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated fadeIn"
  >
    Green Arrow
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="cardGrid"
  />
</Fragment>
`;

exports[`Pruebas en el gifGrid debe mostrar items cuando se cargan imagenes  useFetchGifts 1`] = `
<Fragment>
  <h3
    className="animate__animated fadeIn"
  >
    Green Arrow
  </h3>
  <div
    className="cardGrid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
